crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -637,7 +655,9 @@ pub fn comparator_operand_type_rules(
                 return Err(format!("Can only compare arrays of the same length. Here LHS is of length {}, and RHS is {} ", 
                     x_size, y_size));
             }
-            Ok(Bool)
+
+            // We could check if all elements of all arrays are const but I am lazy
```

Comment:
Can you open up an issue for this? Its not a blocker as we mainly want to have the casting working

Reply:
Done: #280